### PR TITLE
WASM: assert int32 indexes for x64 memory access

### PR DESCRIPTION
**Pull request by Cellule**
OS#14728988
Assert that we can only have int32 indexes for wasm memory operations. 
This is to future proof us in case we start accepting int64 and forget to update this code path.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/4282)

### CODE REVIEW

**Comment by Cellule** [comment]
Ping

File: lib/Backend/amd64/LowererMDArch.cpp [hunk 1]
```diff
@@ -1098,6 +1098,17 @@ LowererMDArch::LowerAsmJsCallI(IR::Instr * callInstr)
 IR::Instr *
 LowererMDArch::LowerWasmMemOp(IR::Instr * instr, IR::Opnd *addrOpnd)
 {
+    IR::IndirOpnd * indirOpnd = addrOpnd->AsIndirOpnd();
+    IR::RegOpnd * indexOpnd = indirOpnd->GetIndexOpnd();
+    if (indexOpnd && !indexOpnd->IsIntegral32())
```
Comment (MikeHolman) [inline lib/Backend/amd64/LowererMDArch.cpp]: do we not want to say uint32?

Comment (Cellule) [reply lib/Backend/amd64/LowererMDArch.cpp]: I think it's possible for Globopt to change the index's type to int32 either through copy-prop or other mechanism

**Comment by Cellule** [comment]
It is indeed overly cautious, but I wouldn't want to have a security bug later that would have been prevented by a runtime check.

**Pull request merged**